Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Follow.cs

```diff
@@ -0,0 +1,644 @@
+            // between toTarget and the leashing bounds about the global Y axis and the reference's X axis.
+            // If toTarget falls within the leashing bounds, then we don't have to modify it.
+            // Otherwise, we apply a correction rotation to bring it within bounds.
+
+            Vector3 currentRefForward = refRotation * Vector3.forward;
+            Vector3 refRight = refRotation * Vector3.right;
+            Vector3 refUp = refRotation * Vector3.up;
+
+            bool angularClamped = false;
+
+            // X-axis leashing
+            // Leashing around the reference's X axis only makes sense if the reference isn't gravity aligned.
+            if (IgnoreReferencePitchAndRoll)
+            {
+                float angle = AngleBetweenOnPlane(toTarget, currentRefForward, refRight);
+                rotation = Quaternion.AngleAxis(angle, refRight) * rotation;
+            }
+            else
+            {
+                float angle = -AngleBetweenVectorAndPlane(toTarget, refUp);
+                float minMaxAngle = MaxViewVerticalDegrees * 0.5f;
+
+                if (angle < -minMaxAngle)
+                {
+                    rotation = Quaternion.AngleAxis(-minMaxAngle - angle, refRight) * rotation;
+                    angularClamped = true;
+                }
+                else if (angle > minMaxAngle)
+                {
+                    rotation = Quaternion.AngleAxis(minMaxAngle - angle, refRight) * rotation;
+                    angularClamped = true;
+                }
+            }
+
+            // Y-axis leashing
+            if (UseAngleStepping)
+            {
+                float stepAngle = 360f / tetherAngleSteps;
+                int numberOfSteps = Mathf.RoundToInt(SolverHandler.TransformTarget.transform.eulerAngles.y / stepAngle);
+
+                float newAngle = stepAngle * numberOfSteps;
+
+                rotation = Quaternion.Euler(rotation.eulerAngles.x, newAngle, rotation.eulerAngles.z);
+            }
+            else
+            {
+                // This is negated because Unity is left-handed
+                float angle = AngleBetweenVectorAndPlane(toTarget, refRight);
+                float minMaxAngle = MaxViewHorizontalDegrees * 0.5f;
+
+                if (angle < -minMaxAngle)
+                {
+                    rotation = Quaternion.AngleAxis(-minMaxAngle - angle, Vector3.up) * rotation;
+                    angularClamped = true;
+                }
+                else if (angle > minMaxAngle)
+                {
+                    rotation = Quaternion.AngleAxis(minMaxAngle - angle, Vector3.up) * rotation;
+                    angularClamped = true;
+                }
+            }
+
+            refForward = rotation * Vector3.forward;
+            return angularClamped;
+        }
+
+        bool DistanceClamp(Vector3 currentPosition, Vector3 refPosition, Vector3 refForward, bool interpolateToDefaultDistance, ref Vector3 clampedPosition)
```

Comment:
It is unclear to me from a quick read what this does. Could you please comment the method?